PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp

```diff
@@ -40,62 +38,171 @@ void GLRenderingDevice::RenderWorld(World* world)
 	for (auto& kv : world->GetWorldComponent<ViewportWorldComponent>()->GetViewports())
 	{
 		// Set viewport rect (TOOO change it to propper rect, not box)
+		CameraComponent* cameraCmp = kv.second.GetCamera();
 		const AARect& rect = kv.second.GetRect();
-		PostprocessSettingsComponent* post = kv.second.GetCamera()->GetSibling<PostprocessSettingsComponent>();
+		const eRenderingModeType renderMode = cameraCmp->GetRenderingMode();
 
 		glViewport((int)(rect.GetMin().X * screenSize.Width), (int)(rect.GetMin().Y * screenSize.Height),
 			(int)(rect.GetSize().X * screenSize.Width), (int)(rect.GetSize().Y * screenSize.Height));
 
-		glDepthMask(GL_FALSE);
-		glDisable(GL_DEPTH_TEST);
+		switch (renderMode)
+		{
+			case eRenderingModeType::UNLIT:
+				RenderUnlit(world, rect, cameraCmp);
+				break;
 
-		if (post && post->UseBgShader)
-			PostprocessRenderingPasses[ePostprocessRenderPassType::BACKGROUND]->Run(world, kv.second.GetCamera(), rect);
-		else
-			PostprocessRenderingPasses[ePostprocessRenderPassType::BACKGROUND_LIGHT]->Run(world, kv.second.GetCamera(), rect);
+			case eRenderingModeType::WIREFRAME:
+				RenderWireframe(world, rect, cameraCmp);
+				break;
 
-		glDepthMask(GL_TRUE);
-		glEnable(GL_DEPTH_TEST);
-		// Render meshes with blin-phong shader
-		GeometryRenderingPasses[eGeometryRenderPassType::BLINN_PHONG]->Run(world, kv.second.GetCamera(), rect);
+			case eRenderingModeType::DEBUG_NORMALS:
+				RenderNormals(world, rect, cameraCmp);
+				break;
 
-		glDepthMask(GL_FALSE);
-		
-		glEnable(GL_BLEND);
+			case eRenderingModeType::LIT:
+			default:
+				RenderLit(world, rect, cameraCmp);
```

Comment:
It's better to invoke assertion when enum is of unknown value.

Reply:
done